src/providers/ogr/qgsgeopackagedataitems.h

```diff
@@ -88,14 +88,18 @@ class QgsGeoPackageCollectionItem : public QgsDataCollectionItem
     //! Returns the layer type from \a geometryType
     static QgsLayerItem::LayerType layerTypeFromDb( const QString &geometryType );
 
-    //! Delete a geopackage layer
+    //! Deletes a geopackage layer
     static bool deleteGeoPackageRasterLayer( const QString &uri, QString &errCause );
 
+
+
   public slots:
 #ifdef HAVE_GUI
     void addTable();
     void addConnection();
     void deleteConnection();
+    //! Compacts (VACUUM) a geopackage database
+    void vacuumGeoPackageDb();
```

Comment:
There is an inconsistency since the implementation of vacuumGeoPackageDb() is not protected by #ifdef HAVE_GUI 
Actually vacuumGeoPackageDb() is ot a GUI thing, so the declaration should be outside #ifdef HAVE_GUI

Reply:
@rouault thanks, I'll fix that